lib/chat_app_web/live/single_room_live.html.heex

```diff
@@ -0,0 +1,92 @@
+            <span class="message_datetime"><%= timestamp %></span>
+            <div class="message_settings_container">
+              <ul class="menu">
+                <li>
+                  <button id={"button-#{message_id}"} class="items-center p-1.5 text-sm font-medium text-center text-gray-900 rounded-lg hover:bg-emerald-200 focus:ring-1 focus:ring-emerald-500 focus:bg-emerald-200" phx-hook="OpenMessageSettings">
+                    <svg id={"svg-#{message_id}"} class="w-6 h-6" aria-hidden="true" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+                      <path id={"path-#{message_id}"} d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"></path>
+                    </svg>
+                  </button>
+                  <div id={"settings-#{message_id}"} hidden class="message_settings_submenu rounded-lg p-1.5 bg-emerald-200 border border-emerald-500" phx-hook="CloseMessageSettings">
+                    <button id={"edit-#{message_id}"} class="hover:bg-emerald-300 rounded-lg p-1.5 text-sm">Edit</button>
+                    <%= if not is_deleted do %>
+                      <button id={"delete-#{message_id}"} class="hover:bg-emerald-300 rounded-lg p-1.5 text-sm">Delete</button>
+                    <% else %>
+                      <button id={"delete-#{message_id}"} disabled class="opacity-25 hover:bg-emerald-300 rounded-lg p-1.5 text-sm">Delete</button>
+                    <% end %>
+                  </div>
+                </li>
+              </ul>
+            </div>
+          </div>
+          <% end %>
+          <div class="message_content m-2">
+            <%= cond do %>
+              <% is_deleted -> %> This message has been deleted.
+              <% true -> %> <%= text %>
+            <% end %>
+          </div>
+        </li>
+      <% end %>
+    <% end %>
+  </ul>
+  <div id="chatbox_anchor"></div>
+</div>
+
+<form action="#" phx-submit="save_message" class="my-2 grid grid-cols-6 gap-2">
+  <textarea name="text" id="message_textarea" maxlength="1000" cols="50" rows="2" class="rounded-2xl bg-green-100 border-0 focus:border focus:border-purple-800 col-span-5 resize-none"></textarea>
+  <button type="submit" class="rounded-2xl text-white text-base bg-purple-800 hover:animate-pulse hover:motion-reduce float-right">Send</button>
+</form>
+
+<div id="emoji_utils_container" class="my-2 gap-2">
+  <button id="emoji_button" class="rounded-2xl px-2 text-white text-base bg-purple-800 hover:animate-pulse hover:motion-reduce">😁</button>
+  <emoji-picker class="light" id="emoji_popup" hidden></emoji-picker>
+  <.back navigate={~p"/rooms"}>Back to room list</.back>
+</div>
+
+<%!--
+<.modal :if={@live_action == :edit} id="room-modal" show on_cancel={JS.patch(~p"/rooms/#{@room}")}>
+    <.live_component
+        module={UltraoriginalChattyChatAppWeb.RoomLive.FormComponent}
+        id={@room.id}
```

Comment:
is the module name correct?
All over the app, the "root module" seems to be "ChatApp".

Reply:
That's correct, it's an old comment from a previous project, I'll delete it.